common_factories.py

```diff
@@ -25,7 +25,7 @@ def run(self):
 
         if master_branch:
             query = """
-            select concat(test_name,',',test_variant) from (select id, test_name,test_variant from test_failure,test_run where branch like '%%%s%%' and test_run_id=id order by test_run_id desc limit %d) x group by test_name,test_variant order by max(id) desc limit %d
+            select concat(test_name,',',test_variant) from (select id, test_name,test_variant from test_failure,test_run where branch='%s' and test_run_id=id order by test_run_id desc limit %d) x group by test_name,test_variant order by max(id) desc limit %d
```

Comment:
For better readability, I suggest using Python's format:
```python
query = f"""
            select concat(test_name,',',test_variant)
            from
              (select id, test_name, test_variant
               from test_failure, test_run
               where
                 branch='{master_branch}' and test_run_id=id
               order by test_run_id desc limit {overlimit}) x
            group by test_name, test_variant
            order by max(id) desc limit {limit}
"""
tests = yield self.mtrDbPool.runQuery(query)
```

Reply:
Also, consider the following changes to make it easier for new contributors to understand the code:
1. using JOIN instead of the equality in the WHERE clause for readability (I don't know by looking at that query which table has column `test_run_id` and which has `id`)
2. table aliases for `test_failure` (such as tf) and `test_run` (such as tr), and specify columns in the select list with:
`tf.id`, `tf.test_name`, `tf.test_variant`.